### PR TITLE
fixes dce regression #10703

### PR DESCRIPTION
**Pull request by cooldome**
frameDefines  are emitted only when they are used

### CODE REVIEW

File: compiler/cgen.nim [hunk 1]
```diff
@@ -539,7 +539,21 @@ proc initLocExprSingleUse(p: BProc, e: PNode, result: var TLoc) =
...
+      FR_.procname = proc; FR_.filename = file; FR_.line = 0; FR_.len = 0; #nimFrame(&FR_);
+
+  $1  define nimfrs_(proc, file, slots, length) \
+      struct {TFrame* prev;NCSTRING procname;NI line;NCSTRING filename; NI len; VarSlot s[slots];} FR_; \
+      FR_.procname = proc; FR_.filename = file; FR_.line = 0; FR_.len = length; #nimFrame((TFrame*)&FR_);
+
+  $1  define nimln_(n, file) \
+      FR_.line = n; FR_.filename = file;
+  """
+  appcg(p.module, p.module.s[cfsForwardTypes], frameDefines, [rope("#")])
```
Comment (Araq) [inline compiler/cgen.nim]: But this should only be added once per C file, not per proc in the C file. Am I missing something?

Comment (cooldome) [reply compiler/cgen.nim]: oops, your are right again ;)
added extra section for frame defines

**Comment by zah** [comment]
Aren't these part of nimbase.h? Why do we emit them in each module as well?

**Comment by Araq** [comment]
Because it fixed some non-trivial bug I don't remember and nimbase.h is on its way out.

**Pull request merged**